disktools.py

```diff
@@ -185,6 +185,8 @@ def readAllInfo(self):
             if name.startswith('/dev/dm-'):
                 n = int(name[8:])
                 pv['pv_name'] = getDeviceMapperNode(n)
+            elif name.startswith('/dev/disk/by-id/dm-name-'):
```

Comment:
The other option here, not sure if it is possible is to `realpath` the name of the PV.

Reply:
Possibly. That only works if what it is being compared against is also using the canonical path.

Deli has a PR here (https://github.com/xenserver/host-installer/pull/250) to fix this issue by comparing the two paths with `samepath()` since the LVM change affects more than just multipath. I'll probably take that change and drop this commit.